openmc/model/model.py

```diff
@@ -1142,38 +1162,63 @@ def differentiate_depletable_mats(self, diff_volume_method: str):
+
+        Parameters
+        ----------
+        diff_volume_method : str
+            Specifies how the volumes of the new materials should be found.
+            Default is to 'divide equally' which divides the original material
+            volume equally between the new materials, 'match cell' sets the
+            volume of the material to volume of the cell they fill.
+        """
+        if diff_volume_method not in ("divide equally", "match cell", None):
+            raise ValueError(
+                "diff_volume_method must be 'divide equally' or 'match cell', "
+                f"not '{diff_volume_method}'"
+            )
         # Count the number of instances for each cell and material
         self.geometry.determine_paths(instances_only=True)
 
         # Extract all depletable materials which have multiple instances
         distribmats = set(
             [mat for mat in self.materials
-                if mat.depletable and mat.num_instances > 1])
+                if (mat.depletable or not depletable_only) and mat.num_instances > 1])
 
-        if diff_volume_method == 'divide equally':
+        if diff_volume_method == "divide equally":
             for mat in distribmats:
                 if mat.volume is None:
-                    raise RuntimeError("Volume not specified for depletable "
-                                        f"material with ID={mat.id}.")
+                    raise RuntimeError(
+                        "Volume not specified for depletable "
+                        f"material with ID={mat.id}."
+                    )
                 mat.volume /= mat.num_instances
 
         if distribmats:
             # Assign distribmats to cells
             for cell in self.geometry.get_all_material_cells().values():
                 if cell.fill in distribmats:
                     mat = cell.fill
-                    if diff_volume_method == 'divide equally':
+                    if diff_volume_method != 'match cell':
                         cell.fill = [mat.clone() for _ in range(cell.num_instances)]
                     elif diff_volume_method == 'match cell':
-                        for _ in range(cell.num_instances):
-                            cell.fill = mat.clone()
+                        cell.fill = mat.clone()
+                        for i in range(cell.num_instances):
                             if not cell.volume:
                                 raise ValueError(
                                     f"Volume of cell ID={cell.id} not specified. "
                                     "Set volumes of cells prior to using "
                                     "diff_volume_method='match cell'."
                                 )
                             cell.fill.volume = cell.volume
+                    if isinstance(cell, openmc.DAGMCCell):
+                        for i in range(cell.num_instances):
+                            cell.fill[i].name = f"{cell.fill[i].name}_{cell.id}_{i}"
```

Comment:
```suggestion
                        for i, f in enumerate(cell.fill):
                            f.name += f"_{cell.id}_{i}"
```

Reply:
Follow-up to this. With the aim of keeping DAGMC-specific code out of otherwise generic blocks of code like this one, do you think it would make sense to populate these fill names during XML element creation for the DAGMC universe?

Reply:
Maybe in the `build_override_mat_from_cells` method?